microshift.py

```diff
@@ -0,0 +1,241 @@
+import time
+import host
+import sys
+from host import Host
+from nfs import NFS
+from logger import logger
+import os
+import shutil
+from typing import Optional
+from jinja2 import Template
+from clustersConfig import NodeConfig
+
+
+# cleans microshift artifacts from previous installation
+def cleanup_microshift(h: host.Host) -> None:
+    files = ["rhocp.toml", "fast-datapath.toml", "kickstart.ks", "*.tar", "*.iso"]
+
+    for file in files:
+        if os.path.exists(f"./{file}"):
+            os.remove(f"./{file}")
+
+    cmd = f"podman rm -f minimal-microshift-server"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    cleanup_compose_cli(h)
+
+    if os.path.exists("/var/cache/osbuild-worker/osbuild-store/stage/"):
+        shutil.rmtree("/var/cache/osbuild-worker/osbuild-store/stage/")
+    return
+
+
+# cleans composes generated by composer cli
+def cleanup_compose_cli(h: host.Host) -> None:
+    old_compose = h.run("composer-cli compose status").out
+    old_compose_splitlines = old_compose.splitlines()
+
+    for line in old_compose_splitlines[1:]:
+        compose_id = line.split(" ")[0]
+        if 'FAILED' or 'FINISHED' not in line:
+            cmd = f" composer-cli compose cancel {compose_id}"
+            output = h.run(cmd)
+            logger.info(output.out)
+        cmd = f"composer-cli compose delete {compose_id}"
+        output = h.run(cmd)
+        logger.info(output.out)
+    return
+
+
+# uses jinja to generate a kickstart file
+def generate_kickstart(rhel_number: str, uname_m: str) -> None:
+    with open('pull_secret.json', 'r') as f_in:
+        file_contents = f_in.read()
+
+    with open('kickstart.ks.j2', 'r') as f:
+        lines = f.read()
+
+    with open('kickstart.ks', 'w') as f_out:
+        template = Template(lines)
+        f_out.write(template.render(rhel_number=rhel_number, uname_m=uname_m, pull_secret=file_contents))
+    return
+
+
+# installs dependencies
+def install_packages(package_name: str, h: host.Host) -> None:
+    cmd = f"sudo dnf install {package_name}"
+    output = h.run(cmd)
+    logger.info(output.out)
+    return
+
+
+# generates non static toml files
+def generate_toml_file(content: str, file_name: str, h: host.Host) -> None:
+    with open(file_name, 'w') as file:
+        file.write(content)
+    return
+
+
+# generates final iso
+def generate_final_iso(file: str, name_of_final_iso: str, h: host.Host) -> None:
+    cmd = f"sudo mkksiso kickstart.ks {file} {name_of_final_iso}"
+    output = h.run(cmd)
+    logger.info(output.out)
+    logger.info("Microshift iso is ready in final.iso")
+    return
+
+
+def change_permissions(file: str) -> None:
```

Comment:
This changes permissions to a specific value of 444. Would there be a better name for this function?